cloud-controller-manager/do/loadbalancers.go

```diff
@@ -1453,6 +1458,26 @@ func getNetwork(service *v1.Service) (string, error) {
 	return network, nil
 }
 
+func getNetworkStack(service *v1.Service, lbType string, network string) (string, error) {
+	networkStack := service.Annotations[annDONetworkStack]
+	if networkStack == "" {
+		if network != godo.LoadBalancerNetworkTypeInternal && lbType != godo.LoadBalancerTypeRegionalNetwork {
+			return godo.LoadBalancerNetworkStackDualstack, nil
+		}
+		return godo.LoadBalancerNetworkStackIPv4, nil
+	}
+	if !(networkStack == godo.LoadBalancerNetworkStackIPv4 || networkStack == godo.LoadBalancerNetworkStackDualstack) {
+		return "", fmt.Errorf("only LB network stacks supported are (%s, %s)", godo.LoadBalancerNetworkStackIPv4, godo.LoadBalancerNetworkStackDualstack)
+	}
+	if lbType == godo.LoadBalancerTypeRegionalNetwork && networkStack == godo.LoadBalancerNetworkStackDualstack {
+		return "", fmt.Errorf("dual stack networking is not supported for regional network LB with kubernetes at this time")
+	}
+	if network == godo.LoadBalancerNetworkTypeInternal && networkStack == godo.LoadBalancerNetworkStackDualstack {
+		return "", fmt.Errorf("dual stack networking is not supported for internal load balancer")
+	}
+	return networkStack, nil
```

Comment:
I feel like the possible combinations are getting quite tricky and easy to mess up for the user.
Would these errors get bubbled up as events to the user on the svc object?